### PR TITLE
[TestProxyMigration-TA] Rerecord all tests

### PR DESCRIPTION
**Pull request by mssfang**
Rerecord all tests in live mode with the new TestProxy.

(Phase 1) Steps to integrate with Test Proxy:
Use azure-core-test newer than 1.15.0. (If there are bug fixes you need you may need to depend on an unreleased version of the lib.)
Change the Base type of the respective SDK test class from TestBase to TestProxyTestBase
Run tests in record mode.
Validate recordings are as expected and do not contain secrets. [Example updated recording](https://github.com/samvaity/azure-sdk-for-java/blob/ed6c25bd9521631080fbd0c2eeeac5180fac4fe4/sdk/formrecognizer/azure-ai-formrecognizer/src/test/resources/session-records/EXAMPLE_DocumentModelAdminClientTest.getModelWithResponse%5B1%5D.json)
Check in the updated recordings to the main repo
Validate changes by running the Live/ci test pipeline

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/src/test/resources/session-records/TextAnalyticsAsyncClientTest.analyzePiiEntityRecognitionWithDomainFilters[1].json [hunk 1]
```diff
@@ -1,67 +1,257 @@
...
+            }
+          }
+        ]
+      },
+      "StatusCode": 202,
+      "ResponseHeaders": {
+        "apim-request-id": "a3dbc43a-09b4-468f-9bcf-01c471424c01",
+        "Content-Length": "0",
+        "Date": "Wed, 05 Apr 2023 18:53:08 GMT",
+        "operation-location": "https://REDACTED/language/analyze-text/jobs/4a7f4fb6-bc33-447b-bcbf-07edb16df598?api-version=2022-10-01-preview",
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/src/test/resources/session-records/TextAnalyticsAsyncClientTest.analyzePiiEntityRecognitionWithDomainFilters[1].json:45]: So happy to finally see this redacted :)

File: sdk/textanalytics/azure-ai-textanalytics/pom.xml [hunk 2]
```diff
@@ -67,7 +67,7 @@
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-core-test</artifactId>
-      <version>1.16.0</version> <!-- {x-version-update;com.azure:azure-core-test;dependency} -->
+      <version>1.17.0-beta.1</version> <!-- {x-version-update;unreleased_com.azure:azure-core-test;dependency} -->
```
[the remark this answers is not in the document]
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/pom.xml:70]: Talked offline. Will revert it with a retry if CI pipeline fails.

File: sdk/textanalytics/azure-ai-textanalytics/src/test/resources/session-records/TextAnalyticsClientBuilderTest.clientBuilderWithDefaultCountryHintForBatchOperation[1].json [hunk 3]
```diff
@@ -1,47 +0,0 @@
-{
```
[the remark this answers is not in the document]
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/test/resources/session-records/TextAnalyticsClientBuilderTest.clientBuilderWithDefaultCountryHintForBatchOperation[1].json:1]: these are disabled tests record files

**Review by samvaity — APPROVED** [review]

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilderTest.java [hunk 4]
```diff
@@ -433,7 +434,13 @@ void clientBuilderWithAadRunner(HttpClient httpClient, TextAnalyticsServiceVersi
                 .pipeline(getHttpPipeline(httpClient))
                 .serviceVersion(serviceVersion);
 
-        if (!interceptorManager.isPlaybackMode()) {
+        if (interceptorManager.isPlaybackMode()) {
+            // since running in playback mode won't have the token credential, so skipping matching it.
+            interceptorManager.addMatchers(Arrays.asList(
+                new CustomMatcher().setExcludedHeaders(Arrays.asList("Authorization"))));
```
[the remark this answers is not in the document]
Comment (samvaity) [reply sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilderTest.java:440]: I think in this case we wanted to set this as with Record we expect the header to be present but with playback we don't set any auth mode at all so are missing even the presence of the header.

File: sdk/textanalytics/azure-ai-textanalytics/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/pom.xml]: removed.

**Comment by billwert** [comment]
:shipit:

**Pull request merged**